scrapy_poet/api.py

```diff
@@ -90,5 +99,12 @@ def parse(self, response):
     def parse(*args, page: page_cls, **kwargs):  # type: ignore
         yield page.to_item()  # type: ignore
 
-    setattr(parse, _CALLBACK_FOR_MARKER, True)
-    return parse
+    async def async_parse(*args, page: page_cls, **kwargs):  # type: ignore
+        yield await page.to_item()  # type: ignore
+
+    if is_async:
```

Comment:
@BurnzZ thanks! Tested, works well.
I'm wondering if it's possible to refactor it this way:
```python
is_async = inspect.iscoroutinefunction(page_cls.to_item)
```
this way it'd be possible to drop `is_async` from function's signature.

Reply:
That's a good point. 👍 I think we can use `inspect.isasyncgenfunction` as well:

```python
import asyncio
import inspect

async def to_item_1():
    yield await asyncio.sleep(1)

async def to_item_2():
    return await asyncio.sleep(1)

inspect.iscoroutinefunction(to_item_1) # False
inspect.iscoroutinefunction(to_item_2) # True

inspect.isasyncgenfunction(to_item_1) # True
inspect.isasyncgenfunction(to_item_2) # False
```

Reply:
Yes, may be `iscoroutinefunction` is not suitable here, my point was about if we can automatically determine whether `to_item` should be awaited or not.

Reply:
Hi @proway2 , it would seem that `iscoroutinefunction` is indeed enough for the conditional check. :) Thanks for the suggestion again! I've updated the code with this.